datashader/core.py

```diff
@@ -167,6 +170,61 @@ def line(self, source, x, y, agg=None):
             agg = any()
         return bypixel(source, self, Line(x, y), agg)
 
+    def raster(self,
+               source,
+               band=1,
+               nodata=np.nan,
+               replace_nodata=None):
+        """Sample a raster dataset by canvas size and bounds. Note: requires
```

Comment:
Another formatting nitpick: top line of docstring should be short (otherwise the sphinx formatting gets wonky). Could put the "Note:..." either in a `Note` section (see this excellent guide to numpydoc standard docstrings: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt), or just in a new paragraph before the parameters bit.